pkg/apis/catalog.cattle.io/v1/types.go

```diff
@@ -111,6 +111,9 @@ type RepoStatus struct {
 	Commit string `json:"commit,omitempty"`
 
 	Conditions []genericcondition.GenericCondition `json:"conditions,omitempty"`
+
+	NumberOfRetries int         `json:"numberOfRetries,omitempty"`
+	NextRetryAt     metav1.Time `json:"nextRetryAt,omitempty"`
```

Comment:
Do we need to have a field for `NextRetryAt`? Maybe we could delegate the tracking and enforcing of this to the underlying controller queue by using `EnqueueAfter`.

What happens in cases that the `NextRetryAt` is not enforced as there was an error before reaching `EnqueueAfter`? Would that not be an invalid state?

Reply:
I'm using `NextRetryAt` because the handler gets triggered after any update, including to the status, so when I update the `NumberOfRetries` on line 167, it gets triggered without waiting for backoff, and then gets triggered again because of the `EnqueueAfter` on line 171

Reply:
Thanks for clarifying.

Effectively we have two sources of truth for when the next reconciliation should take place. It would be nice to have a comment that clarifies that from a high level perspective as otherwise it requires going through several hoops to understand when a reconciliation would take place and when the retry cycle should be reset.

Reply:
Added a comment explaining it